### PR TITLE
Enhance experience + Add more control on the destination directory

### PR DESCRIPTION
**Pull request by yousefhamza**
## Changes
* Fix cocoapods setup
* Move loading remote URLs to viewDidLoad
* Use `lastPathComponent` instead of Title as a header for preview
* Store files in temp cache instead of documents directory
* Expose the URL for remote URL assets

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 Example/Pods
+Example/Podfile.lock
 Example/AMPPreviewControllerExample.xcworkspace/xcuserdata
```
Comment (alexito4) [inline .gitignore]: add xcuserdata to gitignore please.

File: .gitignore [hunk 2]
```diff
@@ -1,2 +1,3 @@
 Example/Pods
+Example/Podfile.lock
```
Comment (alexito4) [inline .gitignore]: the lock file should be committed to ensure the right versions are always installed.

File: Classes/AMPPreviewController.h [hunk 3]
```diff
@@ -43,6 +43,12 @@ typedef void (^AMPPreviewControllerFinishDownload)(NSError *error);
 
 @interface AMPPreviewController : QLPreviewController
 
+/**
+ The URL of the destination directory for the downloaded assets from
+ remoteURL
+ */
+@property (nonatomic, strong, readonly, class) NSURL *tempRemoteURLFilesDirectory;
```
Comment (alexito4) [inline Classes/AMPPreviewController.h:50]: what's the purpose of this?

File: Classes/AMPPreviewController.m [hunk 4]
```diff
@@ -79,12 +83,12 @@ - (id)initWithRemoteFile:(NSURL *)remoteUrl title:(NSString *)title {
     return self;
 }
 
-- (void)viewDidAppear:(BOOL)animated {
```
Comment (alexito4) [inline Classes/AMPPreviewController.m:82]: Could you explain why this change is necessary? 
To be honest I don't remember why I used `viewDidAppear` so many years ago, but I would like to know if the change is required to make it work nowdays.

**Review by alexito4 — CHANGES_REQUESTED** [review]
Thanks for the PR!
As you may imagine this Pod was written long ago and I haven't used it in a long time. I'm inclined to merge it if you can brief me on which OSes you've tested it.

File: .gitignore (hunk 1 above)
Comment (yousefhamza) [reply .gitignore]: Isn't it already added

File: Classes/AMPPreviewController.h (hunk 3 above)
Comment (yousefhamza) [reply Classes/AMPPreviewController.h:50]: Exposing this to external clients so that they can clean it whenever convenient for them

File: Classes/AMPPreviewController.m (hunk 4 above)
Comment (yousefhamza) [reply Classes/AMPPreviewController.m:82]: The issue with this is that it shows a blank screen for a second before showing the actual content (in case the remote file was already cached) which is unnecessary.

By moving it to `viewDidLoad` we get presented with the actual content right away.

**Comment by yousefhamza** [comment]
@alexito4 I have tested it with iOS 13 / 12.

File: .gitignore (hunk 1 above)
Comment (alexito4) [reply .gitignore]: I see config from your user on this PR 
Example/AMPPreviewControllerExample.xcodeproj/project.xcworkspace/xcuserdata/yousefhamza.xcuserdatad/UserInterfaceState.xcuserstate

File: Classes/AMPPreviewController.m (hunk 4 above)
Comment (alexito4) [reply Classes/AMPPreviewController.m:82]: Yes make sense. I'm pretty sure it caused issues in old OSs that's why I moved the code here. But if it's working now I agree `viewDidLoad` is a better place.

File: Classes/AMPPreviewController.h (hunk 3 above)
Comment (alexito4) [reply Classes/AMPPreviewController.h:50]: That makes sense. But to be honest this being iOS we know that what the temp folder is, and we're not making a subdirectory anyway, so we could just document that and let the clients clean NSTemporaryDirectory if they want to. Without making a subdirectory for this Pod I don't see how useful this is, since the client will just endup cleaning the entire temp directory anyway.

**Review by alexito4 — CHANGES_REQUESTED** [review]
Answered your comments. 
Remove Example/AMPPreviewControllerExample.xcodeproj/project.xcworkspace/xcuserdata/yousefhamza.xcuserdatad/UserInterfaceState.xcuserstate please.

File: Classes/AMPPreviewController.h (hunk 3 above)
Comment (yousefhamza) [reply Classes/AMPPreviewController.h:50]: Yes, but I think this should be kept as an implementation detail to give room in the future to improve that and add a proper cache that isn't necessarily the temp directory.

**Review by alexito4 — APPROVED** [review]

**Comment by alexito4** [comment]
Thanks for tackling the PR feedback
